### PR TITLE
Merge v1, v2 DAC and update register definitions

### PR DESCRIPTION
**Pull request by chemicstry**
This merges v1 (unimplemented) and v2 DAC implementations, because they share most of the code except for some exotic stuff, which is not yet implemented for neither of the versions. This should allow using DAC on all chips that have v1 peripheral.

~Currently blocked on https://github.com/embassy-rs/stm32-data/pull/153~

### CODE REVIEW

**Comment by chemicstry** [comment]
So previous DAC implementation used weird `enable` macro instead of `RccPeripheral`, because STM32H7 groups DAC1 and DAC2 into a single enable/reset bit. So instead of manually implementing RCC management for each chip, I refactored DAC to use `RccPeripheral` and added a "hack fix" for H7, since it is the only exception. Is there a better way to fix this, or would this work for now?

File: embassy-stm32/src/dac.rs [hunk 1]
```diff
@@ -1,8 +1,10 @@
+#![macro_use]
```
[the remark this answers is not in the document]
Comment (Dirbaio) [reply embassy-stm32/src/dac.rs:1]: ah nevermind, I see `dac/*.rs` is gone completely.

**Review by Dirbaio — APPROVED** [review]
> Is there a better way to fix this, or would this work for now?

LGTM. RCC bits for analog stuff are cursed either way :) 

bors r+

**Pull request merged**